### DISCUSSION TITLE
Vague error message when using gemgis.visualization.create_dem_3d()

### DISCUSSION THREAD

**Discussion by vdevauxchupin** [cgre-aachen/gemgis#207] [Q&A] [answered]
Hi,

I've been trying to use the function gemgis.visualization.create_dem_3d for one of my DEM but I get the following error:

`ValueError: Input point array shapes must match exactly`

My DEM is imported through rasterio and displays nicely with imshow (see image attached)
![DEM](https://user-images.githubusercontent.com/73993845/149249834-18fef4cc-9536-4677-ad5a-493ef7a4e115.png)
. However I can not manage to create a StructureGrid because of that extent issue.

I tried many things. First, with the extent from the example:

`mesh = gg.visualization.create_dem_3d(dem=dem.read(1), extent=[0,972,0,1069])`

Then, with the extent of the DEM as given by rasterio.bounds: 

`mesh = gg.visualization.create_dem_3d(dem=dem.read(1), extent=[-141.342361111,-139.707083333,59.630694444,60.444861111])`

My issue is that the message that is given is not very clear to me. I don't see in what my DEM is different from the one taken for the example on https://gemgis.readthedocs.io/en/latest/getting_started/tutorial/10_visualizing_data_with_pyvista.html?highlight=read.
What could I do to create a mesh from my DEM ?

Thank you !

**Comment by AlexanderJuestel** [comment, marked as answer]
Hello @vdevauxchupin, 

thanks for your question! For the extent, you will need to use integer values [-141, -140, 60, 60]. These values are used to create the meshgrid which is then used to create the StructuredGrid in Pyvista. So there will be some minor distortions in your raster. Maybe try it and get back to us again :)

Cheers 
Alex

**Reply by vdevauxchupin** [reply]
Hello @AlexanderJuestel ,

Thank you for your answer :) 
Unfortunately your suggestion does not solve the issue, I get the same error message. But I managed to find a way around by using `gg.visualization.read_raster` instead. Thank you though !

**Reply by AlexanderJuestel** [reply]
That's great!
